Tools/CISupport/ews-build/steps.py

```diff
@@ -71,32 +72,24 @@ def getHeaders(self):
         return self._get(self.headers)
 
 
-class GitHubMixin(object):
-    def pr_url(self, pr_number=None):
-        pr_number = pr_number or self.get_pull_request_number()
-        if not pr_number:
-            return ''
-        return '{}/pull/{}'.format(self.getProperty('repository', '-'), pr_number)
-
-    def get_pull_request_number(self):
-        pr_number = self.getProperty('pull_request')
-        if pr_number:
-            return int(pr_number)
+class GitHub(object):
+    @classmethod
+    def repository_urls(cls):
+        return [GITHUB_URL + project for project in GITHUB_PROJECTS]
 
-        if self.getProperty('event') != 'pull_request':
-            return None
-        if self.getProperty('repository') not in GITHUB_REPOSITORIES:
-            return None
+    @classmethod
+    def pr_url(cls, repository_url=None, pr_number=None):
```

Comment:
shouldn't pr_number be mandatory? is there any valid use-case of this method without pr_number?